subsys/shell/backends/Kconfig.backends

```diff
@@ -187,6 +187,101 @@ source "subsys/logging/Kconfig.template.log_config"
 
 endif # SHELL_BACKEND_RTT
 
+config SHELL_BACKEND_MQTT
+	bool "Enable MQTT backend."
+	depends on NET_TCP
+	depends on NET_IPV4
+	depends on NETWORKING
+	select DNS_RESOLVER
+	select HWINFO
+	select MQTT_LIB
+	select NET_MGMT
+	select NET_MGMT_EVENT
+	help
+	  Enable MQTT backend.
+
+if SHELL_BACKEND_MQTT
+
+config SHELL_MQTT_SERVER_ADDR
+	string "MQTT server address"
+	default "192.168.0.100"
+	help
+	  MQTT server address.
+
+config SHELL_MQTT_SERVER_PORT
+	int "MQTT server port"
+	default 1883
+	help
+	  MQTT server port.
+
+config SHELL_MQTT_SERVER_USERNAME
+	string "MQTT server username"
+	help
+	  MQTT server username.
+
+config SHELL_MQTT_SERVER_PASSWORD
+	string "MQTT server password"
+	help
+	  MQTT server password.
+
+config SHELL_MQTT_RX_BUF_SIZE
+	int "RX buffer size"
+	default 256
+	help
+	  Buffer size for the MQTT data reception.
+
+config SHELL_MQTT_TX_BUF_SIZE
+	int "TX buffer size"
+	default 256
```

Comment:
So range would be appropriate here, and possibly also to RX buf size

Reply:
Could you recommend a suitable range? I know that it cant be too short or else the packet will be split into too many packets, but apart from that, the buffer length should be limited only by ram?

Reply:
Well, if you have max 16-bit value to store the length, then 0-65535 is a good range.

Reply:
Understood, thanks

Reply:
If I use `size_t` instead, do you think that range is still necessary?

Reply:
All int sizes are technically correct, it is just that what makes sense here. I think you original uint16_t is a good candidate for the size, I was just commenting that there was a mismatch for the length value and the bit size of the same value and this can lead to issues (perhaps theoretical but still) later.